apis/v1beta1/shared_types.go

```diff
@@ -140,6 +140,13 @@ type CommonRouteSpec struct {
 	// case, the list of routes attached to those resources should also be
 	// merged.
 	//
+	// Note that for ParentRefs that cross namespace boundaries, the parent
+	// being referred to must have some mechanism to allow Routes to attach.
+	// Gateway has the AllowedRoutes field, for example. Any other resource
+	// being used as a parent across namespaces MUST either include an AllowedRoutes
+	// field or equivalent functionality, OR use a ReferenceGrant (if the spec
+	// can't have a field added).
```

Comment:
This is a great clarification, a few nits though:

1. This feels like something that belongs on the Namespace field within the ParentRef type definition.
2. As you mentioned, I think it would be good to include this in both v1a2 and v1b1 type definitions. They should be kept identical.
3. Maybe reword this to say something like "Cross-namespace references are only valid if they are explicitly allowed by something in the namespace they are referring to. For example, Gateway has the AllowedRoutes field, and ReferenceGrant provides a generic way to enable any other kind of cross-namespace reference."

Reply:
Great wording, thanks. I'll update.